src/main/java/org/openstreetmap/atlas/geography/atlas/dynamic/DynamicAtlasExpander.java

```diff
@@ -0,0 +1,691 @@
+        if (!this.policy.isExtendIndefinitely()
+                && !initialShardsBounds.fullyGeometricallyEncloses(location))
+        {
+            // If the policy is to not extend indefinitely, then assume that the loading is not
+            // necessary.
+            return true;
+        }
+        final Iterable<? extends Shard> neededShards = this.sharding.shardsCovering(location);
+        for (final Shard neededShard : neededShards)
+        {
+            if (!this.loadedShards.containsKey(neededShard))
+            {
+                newLocation(location, item);
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Do a preemptive load of the {@link DynamicAtlas} as far as the {@link DynamicAtlasPolicy}
+     * allows.
+     * <p>
+     * In some very specific cases, where the {@link DynamicAtlasPolicy} allows expansion only if
+     * new shards intersect at least one feature that crosses the initial set of shards, it is
+     * possible that expanding only one time misses out some shard candidates. This happens when
+     * some feature intersects the initial shards but does not have any shape point inside any
+     * initial shard. This way, the initial shards do not contain that feature even though they
+     * intersect it. That feature is discovered as we load the neighboring shards which contain that
+     * feature. If that said feature also intersects a third neighboring shard, then that third
+     * neighboring shard becomes eligible for expansion, as that specific feature crosses it and the
+     * initial shards. To work around that case, the preemptive load will do a multi-staged loading.
+     */
+    void preemptiveLoad()
+    {
+        if (!this.policy.isDeferLoading())
+        {
+            logger.warn(
+                    "{}: Skipping preemptive loading as it is useful only when the DynamicAtlasPolicy is deferLoading = true.",
+                    this.dynamicAtlas.getName());
+            return;
+        }
+        if (this.preemptiveLoadDone)
+        {
+            return;
+        }
+        // Loop through the entities to find potential shards to add
+        browseForPotentialNewShards();
+        // Load all the shards into a multiAtlas
+        buildUnderlyingMultiAtlas();
+        // Record the current list of shards
+        Set<Shard> currentShards = new HashSet<>(this.loadedShards.keySet());
+        // Loop through the entities again to find potential shards to add. This can still happen if
+        // a way intersects the initial shard without shapepoints inside the initial shards, and was
+        // revealed by loading a new neighboring shard. At that point, if that way also intersects a
+        // third shard which was not loaded before, that third shard might become now eligible.
+        browseForPotentialNewShards();
+        browseForPotentialNewShardsFromAggressiveRelations();
```

Comment:
This is new, it now browses for potential neighboring shards when the policy has the right parameters.